### DISCUSSION TITLE
WSL2 and DNS resolution failure

### DISCUSSION THREAD

**Discussion by stochmal** [microsoft/WSL#5857] [General]
I am using Ubuntu 20.04 under WSL2 and DNS resolution stopped working. Had to manually add 8.8.8.8 to /etc/resolv.conf to make it work again. No such problem under WSL1

**Comment by craigloewen-msft** [comment]
Hi @stochmal, please keep in mind that this board is for surrounding discussions! Your topic looks somewhat like an issue report to me which is why I point this out. We have different issues on the repo that are tracking DNS/networking issues in WSL 2, like these: https://github.com/microsoft/WSL/issues/5843 , https://github.com/microsoft/WSL/issues/5821 , https://github.com/microsoft/WSL/issues/5791 , and you can expect to see any updates for fixes and news from the engineering side on those issue threads.

Otherwise, please feel free to use this discussion thread for any other related discussion such as workarounds, general questions, etc. 

Thanks!

**Comment by WSLUser** [comment]
A better discussion for WSL2 and DNS. If you enable the systemd hack, systemd-resolved overwrites `/etc/resolve.conf`. How can we get `/etc/wsl.conf` to take priority over the systemd. Do we just disable resolved? I think it might better to keep it but ensure proper nameserver entries are entered as generated by `/etc/wsl.conf`. This is something that needs addressal when https://github.com/microsoft/WSL/issues/994 is addressed.

**Comment by danvy** [comment]
Please, check this PowerShell script and give your feedback
https://github.com/microsoft/WSL/issues/5821#issuecomment-687663258
